File: native/financespy-kmp/shared/src/commonMain/kotlin/py/com/cdco/financespy/db/AccountEntity.kt
package py.com.cdco.financespy.db

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "accounts")
data class AccountEntity(
    @PrimaryKey val id: String,
    val name: String,
    val balanceCents: Long,
    val cashBalanceCents: Long,
    val currency: String,
    val classification: String,
    val accountType: String,
    val subtype: String?,
    val status: String,
    val updatedAt: String
)

File: native/financespy-ios/FinancePY/Persistence/AccountEntity.swift
import Foundation
import SwiftData

@Model
final class AccountEntity {
    @Attribute(.unique) var id: String
    var name: String
    var balanceCents: Int64
    var cashBalanceCents: Int64
    var currency: String
    var classification: String
    var accountType: String
    var subtype: String?
    var status: String
    var updatedAt: String

    init(
        id: String,
        name: String,
        balanceCents: Int64,
        cashBalanceCents: Int64,
        currency: String,
        classification: String,
        accountType: String,
        subtype: String? = nil,
        status: String,
        updatedAt: String
    ) {
        self.id = id
        self.name = name
        self.balanceCents = balanceCents
        self.cashBalanceCents = cashBalanceCents
        self.currency = currency
        self.classification = classification
        self.accountType = accountType
        self.subtype = subtype
        self.status = status
        self.updatedAt = updatedAt
    }
}
